docs/design/features/native-hosting.md

```diff
@@ -333,17 +401,59 @@ Calling this function will load the specified assembly in isolation (into its ow
 ```C
-int component_entry_point_fn(void *arg, int32_t arg_size_in_bytes);
+int get_function_pointer_fn(
+    const char_t *type_name,
+    const char_t *method_name,
+    const char_t *delegate_type_name,
+    void         *load_context,
+    void         *reserved,
+    /*out*/ void **delegate)
 ```
-* `reserved` - parameter reserved for future extensibility, currently unused and must be `0`.
+
+Calling this function will find the specified type in the default load context, locate the required method on it and return a native function pointer to that method. The method's signature can be specified via the delegate type name.
+* `type_name` - Assembly qualified type name to find
+* `method_name` - Name of the method on the `type_name` to find. The method must be `static` and must match the signature of `delegate_type_name`.
+* `delegate_type_name` - Assembly qualified delegate type name for the method signature, or null. If this is null, the method signature is assumed to be:
+    ```C#
+    public delegate int ComponentEntryPoint(IntPtr args, int sizeBytes);
+    ```
+    This maps to native signature:
+    ```C
+    int component_entry_point_fn(void *arg, int32_t arg_size_in_bytes);
+    ```
+    The `delegate_type_name` can be also specified as `UNMANAGEDCALLERSONLY_METHOD` (defined as `(const char_t*)-1`) which means that the managed method is marked with `UnmanagedCallersOnlyAttribute`.
+* `load_context` - eventually this parameter should support specifying which load context should be used to locate the type/method specified in previous parameters. For .NET 5 this parameter must be `NULL` and the API will only locate the type/method in the default load context.
+* `reserved` - parameter reserved for future extensibility, currently unused and must be `NULL`.
 * `delegate` - out parameter which receives the native function pointer to the requested managed method.
 
-It is allowed to call the returned runtime helper many times for different assemblies or different methods from the same assembly. It is not required to get the helper every time. The implementation of the helper will cache loaded assemblies, so requests to load the same assembly twice will load it only once and reuse it from that point onward. Ideally components should not take a dependency on this behavior, which means components should not have global state. Global state in components is typically just cause for problems. For example it may create ordering issues or unintended side effects and so on.
+The helper will lookup the `type_name` from the default load context (`AssemblyLoadContext.Default`) and then return method on it. If the type lookup requires any assemblies to be loaded, they will be resolved and loaded in the default load context.
```

Comment:
Above it says that this is for `already available managed method`, so why are we talking about resolving dependencies? Is the implementation going to use an `AssemblyDependencyResolver` or rely only on the default ALC's normal resolution behavior?

Reply:
The implementation should not use `AssemblyDependencyResolver` as it doesn't have the "component/assembly" to use it on. By the above description I was trying to say that the act of looking up an assembly qualified type name may cause assembly resolution to occur (if the assembly is not yet loaded into the default ALC). Runtime is VERY lazy in loading assemblies and only does so at the last moment it has to. So when the app starts the default ALC will have loaded only the necessary assemblies to get to `Main`, nothing else. If the requested type comes from assemblies outside of that set, the default ALC will perform normal assembly resolution step - in this case almost certainly resolved from TPA and loaded. Getting a method from the type may itself trigger additional assembly resolution/loading (for example if the method has a parameter type from a different assembly) and so on.

I'll try to make this clearer.